### PR TITLE
Test sys light

### PR DESCRIPTION
**Pull request by Liu-Xiaoxiong**
Now everything passed. For 2D Haldane model from tbmodel and pythtb. They need a larger NKFFT to make every *wcc-iter_0000 pass the test. And in order to make *-wcc_iter_0001.dat the same as *_iter_0001.dat, I have to increase NKFFT again. I don't understand why.

### CODE REVIEW

**Comment by Liu-Xiaoxiong** [comment]
New error after merge. 
AttributeError: 'function' object has no attribute 'NKFFT_recommended' 
of system_tb

File: tests/test_berry.py [hunk 1]
```diff
@@ -67,24 +72,40 @@ def compare_quant(quant):
         return quant
 
 
-
-def test_Fe(check_integrate, system_Fe_W90, compare_energyresult,quantities_Fe,Efermi_Fe):
+def test_Fe(check_integrate,system_Fe_W90, compare_energyresult,quantities_Fe,Efermi_Fe):
```
Comment (stepan-tsirkin) [inline tests/test_berry.py:75]: why removing spaces ?

File: tests/test_berry.py [hunk 2]
```diff
@@ -67,24 +72,40 @@ def compare_quant(quant):
...
     """Test anomalous Hall conductivity , ohmic conductivity, dos, cumdos"""
     check_integrate(system_Fe_W90 , quantities_Fe , fout_name="berry_Fe_W90" , suffix="" , Efermi=Efermi_Fe , comparer=compare_energyresult )
 
 
-def test_Fe_wcc(check_integrate, system_Fe_W90_wcc, compare_energyresult,quantities_Fe,Efermi_Fe):
+def test_Fe_wcc(check_integrate,system_Fe_W90_wcc, compare_energyresult,quantities_Fe,Efermi_Fe):
```
Comment (stepan-tsirkin) [inline tests/test_berry.py:80]: generally, lines that are in fact not changes, should not show as changed. Unless you are doing a 'cosmetic' commit.

**Review by stepan-tsirkin — CHANGES_REQUESTED** [review]

File: tests/compare_result.py [hunk 3]
```diff
@@ -47,6 +47,9 @@ def _inner(fout_name, suffix, adpt_num_iter, suffix_ref=None, precision=1E-10):
             path_filename_ref = os.path.join(rootdir, 'reference', filename_ref)
             E_titles_ref, data_energy_ref, data_ref, data_smooth_ref = read_energyresult_dat(path_filename_ref)
 
+            if precision == None:
```
[the remark this answers is not in the document]
Comment (Liu-Xiaoxiong) [reply tests/compare_result.py:50]: Now the default precision is a minimal number of 1E-11 and (average of data_smooth_ref data)/1E12.
```if precision == None:  precision = max(abs(np.average(data_smooth_ref) / 1E12), 1E-11)```
If you want to define a precision by yourself, put it in 'extra_precision'.

File: tests/create_system.py [hunk 4]
```diff
@@ -4,6 +4,8 @@
 import tarfile
 import shutil
 
+import tbmodels
+from pythtb import *
```
[the remark this answers is not in the document]
Comment (Liu-Xiaoxiong) [reply tests/create_system.py]: Sorry!

File: tests/create_system.py [hunk 5]
```diff
@@ -124,3 +126,146 @@ def system_Fe_W90_wcc(create_files_Fe_W90):
...
+            )
+    my_model.add_hop(t, 0, 1, [ 0, 0, 0])
+    my_model.add_hop(t, 1, 0, [ 1, 0, 0])
+    my_model.add_hop(t, 1, 0, [ 0, 1, 0])
+    my_model.add_hop(t2 , 0, 0, [ 1, 0, 0])
+    my_model.add_hop(t2 , 1, 1, [ 1,-1, 0])
+    my_model.add_hop(t2 , 1, 1, [ 0, 1, 0])
+    my_model.add_hop(t2c, 1, 1, [ 1, 0, 0])
+    my_model.add_hop(t2c, 0, 0, [ 1,-1, 0])
+    my_model.add_hop(t2c, 0, 0, [ 0, 1, 0])
```
[the remark this answers is not in the document]
Comment (Liu-Xiaoxiong) [reply tests/create_system.py]: okay!

File: tests/create_system.py [hunk 6]
```diff
@@ -124,3 +126,146 @@ def system_Fe_W90_wcc(create_files_Fe_W90):
...
+
+    return system
+
+
+
+@pytest.fixture(scope="session")
+def system_Haldane_PythTB(rootdir):
+    """Create system for Haldane model using PythTB"""
+    lat=[[1.0,0.0,0.0],[0.5,np.sqrt(3.0)/2.0,0.0],[0.0,0.0,1.0]]
+    orb=[[1./3.,1./3.,0.],[2./3.,2./3.,0.]]
```
[the remark this answers is not in the document]
Comment (Liu-Xiaoxiong) [reply tests/create_system.py]: I did it in 2D but after debugging the mismatch of the Haldane model. I forgot to change it back.

File: tests/test_berry.py [hunk 7]
```diff
@@ -6,13 +6,13 @@
 from pytest import approx
 
 import wannierberri as wberri
-
-from create_system import create_files_Fe_W90,  create_files_GaAs_W90,  system_Fe_W90, system_Fe_W90_wcc,system_GaAs_W90 ,system_GaAs_W90_wcc
+from create_system import *
```
[the remark this answers is not in the document]
Comment (Liu-Xiaoxiong) [reply tests/test_berry.py]: Sorry

File: tests/test_berry.py (hunk 1 above)
Comment (Liu-Xiaoxiong) [reply tests/test_berry.py:75]: I don't know why. There is a  conflict after merge this line.

File: tests/compare_result.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (stepan-tsirkin) [reply tests/compare_result.py]: No, I meant, I do not like ,that those `1e12`,  is hard-coded. I meant smth like 
```
if precision<0 : 
   precision = max(abs(np.average(data_smooth_ref) *(-precision), 1E-11)
```

**Review by stepan-tsirkin — APPROVED** [review]
Great! I had one comment that I fixed myself. You may now "squash and merge"

File: tests/compare_result.py [hunk 8]
```diff
@@ -35,17 +35,24 @@ def error_message(fout_name, suffix, i_iter, abs_err, filename, filename_ref):
...
+        if suffix_ref is None :
+            suffix_ref=suffix
         for i_iter in range(adpt_num_iter+1):
             filename     = fout_name + f"-{suffix}_iter-{i_iter:04d}.dat"
             filename_ref = fout_name + f"-{suffix_ref}_iter-{i_iter:04d}.dat"
             path_filename = os.path.join(output_dir, filename)
             E_titles, data_energy, data, data_smooth = read_energyresult_dat(path_filename)
             path_filename_ref = os.path.join(rootdir, 'reference', filename_ref)
             E_titles_ref, data_energy_ref, data_ref, data_smooth_ref = read_energyresult_dat(path_filename_ref)
+            if precision is None:
```
Comment (stepan-tsirkin) [inline tests/compare_result.py:52]: Ok, but this should be done only once, in the beginning of the function, not at every iteration i_iter

**Pull request merged**